design/comment-scanning.md

```diff
@@ -0,0 +1,110 @@
+# Polymer Analyzer - Comment Directives & Comment Annotations
+> Author @FredKSchott
+> Status: Draft
+> Last Update: 2017-04-25
+
+## Objective
+- To create a paradigm for formatting document metadata and instructions for analysis.
+- Immediately driving this discussion is the need to support polymer-linter directives which can enable/disable lint rules within entire documents and/or single sections of documents.
+
+## Goals
+- Allow the analyzer to surface the information contained within polymer-lint directives so that the linter can properly report & ignore warnings.
+- Create a more general definition for a directive that is easy-to-use for code authors and fits into the data model of the analyzer.
+- Create a more general definition for metadata (ex: @polymerElement) that is attached to an already-existing feature.
+- Documentation (README, docs/ entry, docs site entry, etc) that communicates these new definitions.
+
+## Non-Goals
+- Implement any general-purpose metadata/directive scanners. This document will focus solely on the syntax and the way we talk about scannable documentation going forward.
```

Comment:
If there are no general purpose scanners, then the thing this doc really talks about would be the `Directive` feature. We should have it sketched: what properties does it have?